build.gradle

```diff
@@ -1615,6 +1615,94 @@ void addMavenPublication(Project project, List<String> projectDependencies) {
     }
 }
 
+/**
+ * Verifies that all of the *.java files in a source tree have the
+ * correct package name. If not, fail the build.
+ *
+ * @param sourceRoot the root of the source tree to check
+ * @param hasModuleName true if the sources under sourceRoot are being
+ * compiled as modules, with the immediate child directories being the
+ * root(s) of the named modules being compiled. In this case the package
+ * root is one directory below the sourceRoot.
+ */
+void validatePackages(File sourceRoot, boolean hasModuleName) {
+    if (!sourceRoot.isDirectory()) {
+        return;
+    }
+    def err = false;
+    String sourceString = sourceRoot.toString().replace("\\", "/")
+    def startPos = sourceString.length() + 1
+    logger.info("validating packages for ${sourceRoot}")
+    def inputFiles = fileTree(dir: sourceRoot, include: "**/*.java")
+    inputFiles.each { file ->
+        def packageRoot = file.toString().replace("\\", "/").substring(startPos)
+        if (hasModuleName) {
+            packageRoot = packageRoot.substring(packageRoot.indexOf("/") + 1)
+        }
+        def endPos = packageRoot.lastIndexOf("/")
+        def pkgEx = endPos > -1 ?  packageRoot.substring(0, endPos).replace("/", ".") : ""
+        def pkg = ""
+        file.eachLine { line ->
+            def words = line.trim().split("[ ;]+")
+            if (words.length > 1 && words[0] == "package") {
```

Comment:
I see that it is failing for snippets like below. :) It will be a false positives.
/*
package com.sun.javafx.webkit1.prism;
*/

Reply:
It will also miss the following valid (but unlikely) declaration:

```
/* hi */ package javafx.scene;
```

Which will also cause a false positive. Probably not worth worrying about...